### PR TITLE
Support evaluation for each class separately

### PR DESCRIPTION
**Pull request by linjieccc**
### PR types

New features
### PR changes

APIs
### Description

- Add `separated` for UIE evaluation. Precision, recall and F1 score are calculated for each class separately if `separated` is enabled.

  logging example:
  ```python
  [2022-06-23 08:25:23,017] [    INFO] - -----------------------------
  [2022-06-23 08:25:23,017] [    INFO] - Class name: 时间
  [2022-06-23 08:25:23,018] [    INFO] - Evaluation precision: 1.00000 | recall: 1.00000 | F1: 1.00000
  [2022-06-23 08:25:23,145] [    INFO] - -----------------------------
  [2022-06-23 08:25:23,146] [    INFO] - Class name: 目的地
  [2022-06-23 08:25:23,146] [    INFO] - Evaluation precision: 0.64286 | recall: 0.90000 | F1: 0.75000
  [2022-06-23 08:25:23,272] [    INFO] - -----------------------------
  [2022-06-23 08:25:23,273] [    INFO] - Class name: 费用
  [2022-06-23 08:25:23,273] [    INFO] - Evaluation precision: 0.11111 | recall: 0.10000 | F1: 0.10526
  [2022-06-23 08:25:23,399] [    INFO] - -----------------------------
  [2022-06-23 08:25:23,399] [    INFO] - Class name: 出发地
  [2022-06-23 08:25:23,400] [    INFO] - Evaluation precision: 1.00000 | recall: 1.00000 | F1: 1.00000
  ```
- Support batch prediction for uie_predictor.py

### CODE REVIEW

File: model_zoo/uie/README.md [hunk 1]
```diff
@@ -407,7 +407,7 @@ python doccano.py \
 - ``prompt_prefix``: 声明分类任务的prompt前缀信息，该参数只对分类类型任务有效。默认为"情感倾向"。
 - ``is_shuffle``: 是否对数据集进行随机打散，默认为True。
 - ``seed``: 随机种子，默认为1000.
-- ``seperator``: 实体类别/评价维度与分类标签的分隔符，该参数只对实体/评价维度级分类任务有效。默认为"##"。
+- ``separator``: 实体类别/评价维度与分类标签的分隔符，该参数只对实体/评价维度级分类任务有效。默认为"##"。
```
Comment (wawltor) [inline model_zoo/uie/README.md:410]: 建议改成debug flag

File: model_zoo/uie/evaluate.py [hunk 2]
```diff
@@ -59,22 +59,36 @@ def do_eval():
...
-    test_ds = test_ds.map(
-        partial(convert_example,
-                tokenizer=tokenizer,
-                max_seq_len=args.max_seq_len))
+    class_dict = {}
+    if args.separated:
+        for data in test_ds:
+            class_dict.setdefault(data['prompt'], []).append(data)
+    else:
+        class_dict["All classes"] = test_ds
```
Comment (wawltor) [inline model_zoo/uie/evaluate.py]: 这块建议改成all_classes，符合一下命名规范

File: paddlenlp/taskflow/information_extraction.py [hunk 3]
```diff
@@ -271,16 +274,16 @@ def read(inputs):
                 tokenized_output = [
                     np.array(x, dtype="int64") for x in tokenized_output
                 ]
-
                 yield tuple(tokenized_output)
 
-        infer_ds = load_dataset(read, inputs=short_inputs, lazy=False)
+        infer_ds = load_dataset(read, inputs=short_inputs, lazy=self._lazy_load)
```
Comment (wawltor) [inline paddlenlp/taskflow/information_extraction.py:279]: 这里设置lazy方式在效率上是有提升是吗？

File: model_zoo/uie/README.md (hunk 1 above)
Comment (linjieccc) [reply model_zoo/uie/README.md:410]: done

File: model_zoo/uie/evaluate.py (hunk 2 above)
Comment (linjieccc) [reply model_zoo/uie/evaluate.py]: done

File: paddlenlp/taskflow/information_extraction.py (hunk 3 above)
Comment (linjieccc) [reply paddlenlp/taskflow/information_extraction.py:279]: 这里默认是False，增加可配置项防止用户使用Taskflow批量预测时传入大量数据一次性加载占用过多内存

**Review by wawltor — DISMISSED** [review]
LGTM

**Review by wawltor — APPROVED** [review]
LGTM

**Pull request merged**